pkg/authorization/workspace_content_authorizer.go

```diff
@@ -100,91 +101,89 @@ func (a *workspaceContentAuthorizer) Authorize(ctx context.Context, attr authori
 		&rbac.ClusterRoleBindingLister{Lister: parentWorkspaceKubeInformer.ClusterRoleBindings().Lister()},
 	)
 
-	// TODO: decide if we want to require workspaces for all kcp variations. For now, only check if the workspace controllers are running,
-	// as that ensures the ClusterWorkspace CRD is installed, and that our shared informer factory can sync all its caches successfully.
-	if a.clusterWorkspaceLister != nil {
-		// check the workspace even exists
-		// TODO: using scoping when available
-		if ws, err := a.clusterWorkspaceLister.Get(clusters.ToClusterAwareKey(parentClusterName, clusterWorkspace)); err != nil {
-			if errors.IsNotFound(err) {
-				return authorizer.DecisionDeny, fmt.Sprintf("%q workspace access not permitted", cluster.Name), nil
-			}
-			return authorizer.DecisionNoOpinion, "", err
-		} else if len(ws.Status.Initializers) > 0 {
+	extraGroups := sets.NewString()
+
+	// check the workspace even exists
+	if ws, err := a.clusterWorkspaceLister.Get(clusters.ToClusterAwareKey(parentClusterName, clusterWorkspace)); err != nil {
+		if errors.IsNotFound(err) {
+			return authorizer.DecisionDeny, fmt.Sprintf("%q workspace access not permitted", cluster.Name), nil
+		}
+		return authorizer.DecisionNoOpinion, "", err
+	} else if len(ws.Status.Initializers) > 0 {
+		// service accounts can potentially access sub-workspaces, but not other workspaces during initialization. Hence, these two case:
+		// 1. it's  not a serivce account
+		// 2. it's a service account, and it is defined in the parent workspace
+		if len(subjectClusters) == 0 && subjectClusters[parentClusterName] {
 			workspaceAttr := authorizer.AttributesRecord{
 				User:            attr.GetUser(),
-				Verb:            attr.GetVerb(),
+				Verb:            "initialize",
```

Comment:
what is the axiom by which we evaluate the previous (subresource=initialize) to this proposed (verb=initialize) approach?